main/src/library/MutSet.flix

```diff
@@ -338,22 +344,22 @@ namespace MutSet {
     ///
     /// The function `f` must be pure.
     ///
-    pub def partition(f: a -> Bool, s: MutSet[a]): (MutSet[a], MutSet[a]) & Impure with Order[a] =
+    pub def partition(f: a -> Bool, s: MutSet[a, r]): (MutSet[a, Impure], MutSet[a, Impure]) & Impure with Order[a] =
```

Comment:
I would have the result be in `r`. Thus the effect would not be impure but r.

Reply:
We cannot do this since `partition` introduces two new references and does not take a region parameter, but we can easily refactor this.

Reply:
... or add a dirty cast `(MutSet1(ref ys), MutSet1(ref zs)) as (MutSet1[a, r], MutSet1[a, r]) & reg(r)`

Reply:
I see okay. Ill think about it.